File: java/monte.java
import java.util.Random;

public class monte {
    public static void main(String[] args) {
        if (args.length < 1) {
            System.err.println("Usage: java MonteCarloPi <iterations>");
            System.exit(1);
        }

        long n = Long.parseLong(args[0]);
        Random rng = new Random(42);  // LCG с фиксированным seed (аналог mt19937_64 по семантике использования)

        long inside = 0;
        for (long i = 0; i < n; i++) {
            double x = rng.nextDouble();  // [0.0, 1.0)
            double y = rng.nextDouble();  // [0.0, 1.0)
            if (x * x + y * y <= 1.0) {
                inside++;
            }
        }

        System.out.println(4.0 * inside / n);
    }
}

File: ocaml/monte.ml
let estimate_pi n =
  (* Создаем генератор с фиксированным seed = 42 *)
  let rng = Random.State.make [|42|] in
  let inside = ref 0 in
  
  for i = 1 to n do
    let x = Random.State.float rng 1.0 in
    let y = Random.State.float rng 1.0 in
    if (x *. x) +. (y *. y) <= 1.0 then
      incr inside
  done;
  
  4.0 *. (float_of_int !inside) /. (float_of_int n)

let main () =
  if Array.length Sys.argv < 2 then
    (Printf.eprintf "Usage: %s <number_of_iterations>\n" Sys.argv.(0); exit 1)
  else
    let n = int_of_string Sys.argv.(1) in
    let pi_estimate = estimate_pi n in
    Printf.printf "π ≈ %.10f\n" pi_estimate;
    Printf.printf "Error: %.10f\n" (abs_float (Float.pi -. pi_estimate))

let () = main ()